### PR TITLE
Added DeltaExtras class and Message#delta_extras method. (TM2i)

### LINKED ISSUE #273: Add support for TM2i (DataTypes Message)
Add support for the following requirements: TM2i

As stated at https://docs.ably.io/client-lib-development-guide/features/#TM2i

see details https://github.com/ably/docs/blob/main/content/client-lib-development-guide/versions/features-1-1__1-2.diff

Acceptance Criteria: 

Implement the support for the requirements above

Update the spreadsheet

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-1077) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by lukaszsliwa**
Resolves #273

Added `DeltaExtras` class and `Message#delta_extras`

### CODE REVIEW

**Review by owenpearson — APPROVED** [review]
LGTM, thanks

File: lib/ably/models/delta_extras.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  class DeltaExtras
+    include Ably::Modules::ModelCommon
+
+    # DeltaExtras attributes
+    # @return [Hash]
+    #
+    attr_reader :attributes
+
+    def initialize(attributes = {})
+      @attributes = IdiomaticRubyWrapper((attributes || {}), stop_at: [:from, :format])
```
Comment (TheSmartnik) [inline lib/ably/models/delta_extras.rb:17]: What do you think about? 
```
attr_reader :from, :format

@from, @format = IdiomaticRubyWrapper((attributes || {}), stop_at: [:from, :format]). values_at(:from, :format)
```

It's way shorter and doesn't introduce and expose `attributes`

**Review by TheSmartnik — CHANGES_REQUESTED** [review]
Could you please add unit specs for `DeltaExtras` and a small method spec for `Message`?

File: lib/ably/models/delta_extras.rb (hunk 1 above)
Comment (lukaszsliwa) [reply lib/ably/models/delta_extras.rb]: Fixed

**Comment by lukaszsliwa** [comment]
> Could you please add unit specs for `DeltaExtras` and a small method spec for `Message`?

Yes, added these tests.

**Review by TheSmartnik — APPROVED** [review]

**Pull request merged**